commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/PaperCommandRegistration.java

```diff
@@ -89,9 +157,77 @@ public void unregister(String commandName, boolean unregisterNamespaces, boolean
 
 	@Override
 	public void preReloadDataPacks() {
-		RootCommandNode<Source> root = getBrigadierDispatcher.get().getRoot();
+		RootCommandNode<Source> root = getPaperDispatcher().getRoot();
 		for (CommandNode<Source> commandNode : registeredNodes.getChildren()) {
 			root.addChild(commandNode);
 		}
+		reloadHelpTopics.run();
+		CommandAPIBukkit.get().updateHelpForCommands(CommandAPI.getRegisteredCommands());
 	}
+
+	@SuppressWarnings("unchecked")
+	private LiteralCommandNode<Source> asPluginCommand(LiteralCommandNode<Source> commandNode) {
+		try {
+			if (pluginCommandNodeConstructor.getDeclaringClass().getSimpleName().equals("PluginCommandNode")) {
+				return (LiteralCommandNode<Source>) pluginCommandNodeConstructor.newInstance(
+					commandNode.getLiteral(),
+					CommandAPIBukkit.getConfiguration().getPlugin().getPluginMeta(),
+					commandNode,
+					getDescription(commandNode.getLiteral())
+				);
+			} else {
+				setPluginCommandMeta(commandNode);
+				return commandNode;
+			}
+		} catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private void setPluginCommandMeta(LiteralCommandNode<Source> node) {
+		try {
+			Field metaField = node.getClass().getSuperclass().getDeclaredField("pluginCommandMeta");
+			metaField.setAccessible(true);
+			metaField.set(node, pluginCommandNodeConstructor.newInstance(
+				CommandAPIBukkit.getConfiguration().getPlugin().getPluginMeta(),
+				getDescription(node.getLiteral()),
+				getAliasesForCommand(node.getLiteral())
+			));
+		} catch (NoSuchFieldException | InvocationTargetException | InstantiationException | IllegalAccessException e) {
+			// This doesn't happen
+		}
+	}
+
+	private String getDescription(String commandName) {
+		String description = "";
+		for (RegisteredCommand command : CommandAPI.getRegisteredCommands()) {
+			String namespaceStripped = "";
+			if (commandName.contains(":")) {
+				namespaceStripped = commandName.split(":")[1];
+			} else {
+				namespaceStripped = commandName;
+			}
```

Comment:
We don't need to compute `namespacedStripped` on every iteration of this loop. It only depends on `commandName`, which doesn't change.
```suggestion
		String namespaceStripped = "";
		if (commandName.contains(":")) {
			namespaceStripped = commandName.split(":")[1];
		} else {
			namespaceStripped = commandName;
		}
		for (RegisteredCommand command : CommandAPI.getRegisteredCommands()) {
```